mods/flowers/init.lua

```diff
@@ -15,113 +15,50 @@ minetest.register_alias("flowers:flower_rose", "flowers:rose")
-	selection_box = {
-		type = "fixed",
-		fixed = { -0.15, -0.5, -0.15, 0.15, 0.2, 0.15 },
-	},
-})
+-- Flower registration function
+local function add_simple_flower(name, desc, f_groups)
+	-- Common flowers' groups
+	f_groups.snappy = 3
+	f_groups.flammable = 2
+	f_groups.flower = 1
+	f_groups.flora = 1
+	f_groups.attached_node = 1
 
-minetest.register_node("flowers:rose", {
-	description = "Rose",
-	drawtype = "plantlike",
-	tiles = { "flowers_rose.png" },
-	inventory_image = "flowers_rose.png",
-	wield_image = "flowers_rose.png",
-	sunlight_propagates = true,
-	paramtype = "light",
-	walkable = false,
-	buildable_to = true,
-	groups = {snappy=3,flammable=2,flower=1,flora=1,attached_node=1,color_red=1},
-	sounds = default.node_sound_leaves_defaults(),
-	selection_box = {
-		type = "fixed",
-		fixed = { -0.15, -0.5, -0.15, 0.15, 0.3, 0.15 },
-	},
-})
+	minetest.register_node("flowers:"..name.."", {
+		description = desc,
+		drawtype = "plantlike",
+		tiles = { "flowers_" .. name .. ".png" },
+		inventory_image = "flowers_" .. name .. ".png",
+		wield_image = "flowers_" .. name .. ".png",
+		sunlight_propagates = true,
+		paramtype = "light",
+		walkable = false,
+		stack_max = 99,
+		groups = f_groups,
+		sounds = default.node_sound_leaves_defaults(),
+		selection_box = {
+			type = "fixed",
+			fixed = { -0.15, -0.5, -0.15, 0.15, 0.2, 0.15 },
+		},
+	})
+end
 
-minetest.register_node("flowers:tulip", {
-	description = "Tulip",
-	drawtype = "plantlike",
-	tiles = { "flowers_tulip.png" },
-	inventory_image = "flowers_tulip.png",
-	wield_image = "flowers_tulip.png",
-	sunlight_propagates = true,
-	paramtype = "light",
-	walkable = false,
-	buildable_to = true,
-	groups = {snappy=3,flammable=2,flower=1,flora=1,attached_node=1,color_orange=1},
-	sounds = default.node_sound_leaves_defaults(),
-	selection_box = {
-		type = "fixed",
-		fixed = { -0.15, -0.5, -0.15, 0.15, 0.2, 0.15 },
-	},
-})
+-- Registrations using the function above
+flowers.datas = {
+	{"dandelion_yellow", "Yellow Dandelion", {color_yellow=1}},
+	{"geranium", "Blue Geranium", {color_blue=1}},
+	{"rose", "Rose", {color_red=1}},
+	{"tulip", "Orange Tulip", {color_orange=1}},
+	{"dandelion_white", "White dandelion",{color_white=1}},
+	{"viola", "Viola",  {color_violet=1}}
+}
 
-minetest.register_node("flowers:viola", {
-	description = "Viola",
-	drawtype = "plantlike",
-	tiles = { "flowers_viola.png" },
-	inventory_image = "flowers_viola.png",
-	wield_image = "flowers_viola.png",
-	sunlight_propagates = true,
-	paramtype = "light",
-	walkable = false,
-	buildable_to = true,
-	groups = {snappy=3,flammable=2,flower=1,flora=1,attached_node=1,color_violet=1},
-	sounds = default.node_sound_leaves_defaults(),
-	selection_box = {
-		type = "fixed",
-		fixed = { -0.5, -0.5, -0.5, 0.5, -0.2, 0.5 },
-	},
-})
+for _,item in pairs(flowers.datas) do
+	name = item[1]
+	description = item[2]
+	groups = item[3]
+	add_simple_flower(name, description, groups)
```

Comment:
It does, and also thanks, you taught me a new (useful) method.
https://github.com/LeMagnesium/minetest_game/commit/5de52e242a9b9e096d879ae926d8b647773e8f5c